src/Function.cpp

```diff
@@ -875,6 +875,14 @@ GenerateFunctions(void)
 			ERROR_RETURN();
 		}
 	}
+	if(CGOptions::trans_memory_atomic()){
+		for (int i = 0; i < FuncListSize(); i++) {
+			if(rnd_flipcoin(TransAtomicProb)){
+				if (!FuncList[i]->blocks[0]->atomic_blk)
+						FuncList[i]->blocks[0]->atomic_blk = true;
```

Comment:
Still I don't like this.

Reply:
Should I remove all of above? I did not understand what you want.

Reply:
Yes, remove it and make the random decision in `Block::Output`.